drivers: spi: Fix DMA overflow in NXP MCUX driver

mmahadevan108 opened the pull request:
The dummy buffer is not allocated to match the amount of data that is transferred. This results in DMA transfer overflowing to overwrite memory. This resulted in SPI test failures with DMA.

teburd commented on drivers/spi/spi_mcux_flexcomm.c:
```diff
@@ -631,12 +631,19 @@ static int transceive_dma(const struct device *dev,
 		data->status_flags = 0;
 
+		if (data->ctx.rx_buf == NULL) {
+			dummy_rx_buffer = k_malloc(dma_len);
```
Shouldn't this not be needed if the DMA is setup to not increment the address of the receive when using the dummy word?

mmahadevan108 replied:
Great point @teburd and thank you for your review comment.
On further investigation I did come across for a similar feature in this controller as well. I have updated the PR.

teburd reviewed (changes requested):
This seems wrong, edma supports not-incrementing the address? This seems like maybe a misconfiguration when using the dummy rather than requiring a temporary buffer to receive into like this patch does

mmahadevan108 committed:
drivers: spi: Fix DMA overflow in NXP MCUX driver

When using a dummy buffer on the RX side, do not
increment the destination memory buffer address.
This issue was uncovered when running the SPI
loopback test.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

teburd commented on drivers/spi/spi_mcux_flexcomm.c:
```diff
@@ -519,6 +519,8 @@ static int spi_mcux_dma_rx_load(const struct device *dev, uint8_t *buf,
 	if (buf == NULL) {
 		/* if rx buff is null, then write data to dummy address. */
 		blk_cfg->dest_address = (uint32_t)&dummy_rx_buffer;
+		blk_cfg->source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
```
This seems like the correct option regardless of whether the dummy or not is being used

mmahadevan108 replied:
I have removed this line as source increment is already 0 based on the transfer type (PERIPHERAL to MEMORY). I had to update the DMA driver as it was looking at `source and destination addr_adj` elements together.

teburd replied:
makes sense to me

mmahadevan108 committed:
drivers: dma: mcux_lpc: Update address increment logic

The address increment logic was looking at both source
and destination parameters together. Separate them so
we can set one.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>
